### ISSUE TITLE
Unable to Preview Previous Chat in Copilot Chat

### ISSUE THREAD

**Issue by Sophyia7** [pieces-app/example-ts#59]
### Problem Statement
Currently, there is an issue where you can't preview the previous chat in Copilot Chat. This impacts the user's ability to review or refer to earlier parts of the conversation.

**Expected Behavior:**
Users should be able to preview and reference the previous chat messages easily.

**Comment by RohanMishra315** [comment]
Hey @Sophyia7 Can I take a look ?

**Comment by jordan-pieces** [comment]
@RohanMishra315 once you submit your PR on the activities endpoint this can be your next one if you would like to 👍 let us know following that PR being opened.

**Comment by RohanMishra315** [comment]
@jordan-pieces agrim will do the PR of activities endpoint . I am up for this , let's discuss

**Comment by jordan-pieces** [comment]
hey @RohanMishra315 this is great, let me know what you may need or any further conversation that you would like to have 👍

**Comment by RohanMishra315** [comment]
Hey @jordan-pieces I was asking the implementation should happen in here src/app/components/Copilot/Copilot.tsx right ?

**Comment by jordan-pieces** [comment]
@RohanMishra315 yes that should be the proper location to work in the logic of the previous chat. You can use some of the information here found here on line `122` inside of the getInitialChat() method:

```
 await new Pieces.ConversationsApi().conversationsSnapshot({}).then(output => {
       // this is inside of the current conversation, and the first conversation message is what we are looking for.
       // another value here that is important is the ID of the conversation, that is stored as the globalConversationID 
       // that can be used to reference this conversation. each message sent here is appended to the list.
    
        _name = output.iterable.at(0).name;
        GlobalConversationID = output.iterable.at(0).id;
        return output.iterable.at(0).name
      });
```

With that info you should start to see the previous chats that are still present in the list and display them when the UI is updated via `UseEffect()`.

**Comment by shagunsharma6677** [comment]
Hey @jordan-pieces, could you please assign this issue to me if no one is working on it? I have spent enough time understanding the issue, and I just want to give it a shot if possible!

Thank you.

**Comment by jordan-pieces** [comment]
@RohanMishra315 do you have an update here? would love to get some progress or we can get this assigned to @shagunsharma6677 here while you finish up any other work. let me know 👍

**Comment by RohanMishra315** [comment]
I have been busy lately, I had done the changes ,will review and do the PR soon.Thanks @shagunsharma6677 for showing interest.

**Comment by jordan-pieces** [comment]
Awesome looking forward to it @RohanMishra315 and hope you are well. will keep an eye out
